ote/test/clj/ote/transit_changes/detection_test.clj

```diff
@@ -169,10 +173,302 @@
+         {route-name ["h1" "!!" "!!" "h4" "h5" "h6" "h7"]} ;; no changes here (25.2.)
+         {route-name ["h1" "h2" "!!" "h4" "h5" "h6" "h7"]} ;; Only tuesday is found (4.3.)
+         {route-name ["h1" "h2" "!!" "h4" "h5" "h6" "h7"]} ;;
+         {route-name ["h1" "h2" "!!" "h4" "h5" "h6" "h7"]} ;;
+         {route-name ["h1" "h2" "!!" "h4" "h5" "h6" "h7"]}))
+
+
+(def test-more-than-one-change-2-routes
+  ; Produce change records about individual days -> first change week contains 2 days with differences
+  ; In this test case we need to produce 3 rows in the database
+  (weeks (d 2019 2 4)
+         {route-name ["h1" "h2" "h3" "h4" "h5" "h6" "h7"]
+        route-name-2 ["h1" "h2" "h3" "h4" "h5" "h6" "h7"]} ;; 4.2.
+         {route-name ["h1" "h2" "h3" "h4" "h5" "h6" "h7"]
+        route-name-2 ["h1" "h2" "h3" "h4" "h5" "h6" "h7"]} ;; first current week (11.2.)
+         {route-name ["h1" "!!" "!!" "h4" "h5" "h6" "h7"]
+        route-name-2 ["h1" "!!" "!!" "h4" "h5" "h6" "h7"]} ;; 18.2. first change -> only one found currently | needs to detect change also on 13.2. -> this is set to current week
+         {route-name ["h1" "!!" "!!" "h4" "h5" "h6" "h7"]
+        route-name-2 ["h1" "!!" "!!" "h4" "h5" "h6" "h7"]} ;; no changes here (25.2.)
+         {route-name ["h1" "h2" "!!" "h4" "h5" "h6" "h7"]
+        route-name-2 ["h1" "h2" "!!" "h4" "h5" "h6" "h7"]} ;; Only tuesday is found (4.3.)
+         {route-name ["h1" "h2" "!!" "h4" "h5" "h6" "h7"]
+        route-name-2 ["h1" "h2" "!!" "h4" "h5" "h6" "h7"]} ;;
+         {route-name ["h1" "h2" "!!" "h4" "h5" "h6" "h7"]
+        route-name-2 ["h1" "h2" "!!" "h4" "h5" "h6" "h7"]} ;;
+         {route-name ["h1" "h2" "!!" "h4" "h5" "h6" "h7"]
+        route-name-2 ["h1" "h2" "!!" "h4" "h5" "h6" "h7"]}))
+
+(def data-two-week-change
+  (weeks (d 2019 2 4)
+         {route-name ["h1" "h2" "h3" "h4" "h5" "h6" "h7"]} ;; 4.2.
+         {route-name ["h1" "h2" "h3" "h4" "h5" "h6" "h7"]} ;; first current week (11.2.)
+         {route-name ["h1" "!!" "!!" "h4" "h5" "h6" "h7"]}
+         {route-name ["h1" "!!" "!!" "h4" "h5" "h6" "h7"]}
+         {route-name ["h1" "h2" "h3" "h4" "h5" "h6" "h7"]}
+         {route-name ["h1" "h2" "h3" "h4" "h5" "h6" "h7"]}
+         {route-name ["h1" "h2" "h3" "h4" "h5" "h6" "h7"]}
+         {route-name ["h1" "h2" "h3" "h4" "h5" "h6" "h7"]}))
+
+(deftest more-than-one-change-found
+  (spec-test/instrument `detection/route-weeks-with-first-difference)
+
+  ;; first test that the test data and old change detection code agree
+  (testing "single-change detection code agrees with test data"
+    (is (= (d 2019 2 18) (-> test-more-than-one-change
+                             detection/route-weeks-with-first-difference
+                             (get route-name)
+                             :different-week
+                             :beginning-of-week))))
+  
+  (let [diff-pairs (-> test-more-than-one-change
```

Comment:
general "diff-pairs" naming comment, the data format isn't pairs anymore I think, so should globally rename these